src/robotFactory.js

```diff
@@ -1,10 +1,72 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
```suggestion
  constructor(name, weight, coords = {}, chipVersion) {
```
if coords are not provided at all an error will be thrown when you try to read `coords.x` or `coords.y` below